app/src/main/java/com/example/mytestproject/data/repository/WeatherRepositoryModule.kt

```diff
@@ -0,0 +1,22 @@
+package com.example.mytestproject.data.repository
+
+import com.example.mytestproject.data.network.ApiService
+import com.example.mytestproject.mapp.mappWeatherDasta.WeatherDataMapper
+import dagger.Module
+import dagger.Provides
+
+@Module
+class WeatherRepositoryModule {
+
+    @Provides
+    fun provideWeatherApi() = ApiService()
```

Comment:
Usually we keep only one instance of Repository/API/DB objects, And this instance is used everywhere in the app (if needed).
Let's put ApiService and WeatherRepository in Singleton scope.